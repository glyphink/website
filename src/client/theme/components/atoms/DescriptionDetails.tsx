/**
 * @file Components - DescriptionDetails
 * @module website/theme/components/atoms/DescriptionDetails
 */

import type { Props } from '@glyphink/website/theme'
import {
  forwardRef,
  type ForwardedRef,
  type FRC,
  type JSX
} from 'react'
import type DescriptionDetailsProps from './DescriptionDetails.props.mts'

/**
 * The description, definition, or value for the preceding term (`<dt>`) in a
 * description list (`<dl>`).
 *
 * > ðŸ‘‰ **Note**: Must be used inside a `DescriptionList` component.
 *
 * @see https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/dd
 * @see {@linkcode DescriptionDetailsProps}
 * @see {@linkcode FRC}
 *
 * @category
 *  atoms
 *
 * @const {FRC<DescriptionDetailsProps>} DescriptionDetails
 */
const DescriptionDetails: FRC<DescriptionDetailsProps> = forwardRef<
  HTMLElement,
  DescriptionDetailsProps
>(
  /**
   * @param {Props<DescriptionDetailsProps>} props
   *  Component props
   * @param {ForwardedRef<HTMLElement>} ref
   *  Forwarded ref
   * @return {JSX.Element}
   *  JSX `<dd>` element
   */
  (
    props: Props<DescriptionDetailsProps>,
    ref: ForwardedRef<HTMLElement>
  ): JSX.Element => {
    return <dd {...props} ref={ref} />
  }
)

DescriptionDetails.displayName = 'DescriptionDetails'

export default DescriptionDetails
