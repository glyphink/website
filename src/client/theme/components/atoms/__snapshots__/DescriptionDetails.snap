// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:theme/components/atoms/DescriptionDetails > should render <dd> 1`] = `
<dd>
  A large feline inhabiting Bodmin Moor.
</dd>
`;
